src/EnergyPlus/EarthTube.cc

```diff
@@ -152,12 +157,53 @@ void GetEarthTube(EnergyPlusData &state, bool &ErrorsFound) // If errors found i
     // Following used for reporting
     state.dataEarthTube->ZnRptET.allocate(state.dataGlobal->NumOfZones);
 
-    constexpr std::string_view cCurrentModuleObject = "ZoneEarthtube";
-    int TotEarthTube = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+    std::string_view cCurrentModuleObject = "ZoneEarthtube:Parameters";
+    int totEarthTubePars = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+
+    state.dataEarthTube->EarthTubePars.allocate(totEarthTubePars);
+
+    for (Loop = 1; Loop <= totEarthTubePars; ++Loop) {
+        auto &thisEarthTubePars = state.dataEarthTube->EarthTubePars(Loop);
```

Comment:
Good use of reference here, this section is nice and clean.

Reply:
Thanks--I tried to keep this as clean as possible and even converted some of my early work so that it was part of the struct to avoid passing some things and be able to use the this-> approach.  I'm still learning but hopefully this benefited from some of your investment in explaining various things to me over the last couple of years. 😀